tests/errors.spec.ts

```diff
@@ -0,0 +1,43 @@
+/// <reference path="mochainon.d.ts" />
+
+import 'mocha';
+import * as m from 'mochainon';
+import * as errors from '..';
+
+type ResinErrorClass = typeof errors.ResinError;
+
+const runForAllErrors = (fn: (errorName: string, ErrorClass: ResinErrorClass) => void) => {
+	for (const errorName in errors) {
+		if (errors.hasOwnProperty(errorName)) {
+			fn(errorName, errors[errorName] as ResinErrorClass);
```

Comment:
This is upset because `errorName` is just a string, and `errors` doesn't have an index signature, so it has no idea what the result of searching it for a string is (what's the type of `errors[unknownString]`?)

The missing info it needs is that `errorName` is a key of `errors` - annoyingly for/in isn't enough to know that in general: https://github.com/Microsoft/TypeScript/issues/12314.

We are pretty sure of that ourselves though, so we can just cast it: `errors[errorName as keyof typeof errors]`.

That works, but leaves you with two casts (this new key cast, and then the `ResinErrorClass` cast as well). That's a little messy, but it's not actually safe to remove `as ResinErrorClass`, because that isn't actually true. Part of the type of ResinErrorClass is that it has a constructor which takes an error as the first argument. Lots of these error classes don't have that, so don't match the type. Type hierarchies of class objects (as opposed to instances) are weird. What you want to say it that it's an object with the same prototype, not that it's literally a same-shaped class (because it's not).